aws_lambda_powertools/utilities/parameters/secrets.py

```diff
@@ -224,3 +359,88 @@ def get_secret(
         force_fetch=force_fetch,
         **sdk_options,
     )
+
+
+def set_secret(
+    name: str,
+    value: Union[str, bytes],
+    *,  # force keyword arguments
+    client_request_token: Optional[str] = None,
+    **sdk_options,
+) -> str:
+    """
+    Modify the details of a secret or create a new secret if it doesn't already exist.
+    It includes metadata and the secret value.
+
+    We aim to minimize API calls by assuming that the secret already exists and needs updating.
+    If it doesn't exist, we attempt to create a new one. Refer to the following workflow for a better understanding:
+
+
+                      ┌────────────────────────┐      ┌─────────────────┐
+            ┌───────▶│Resource NotFound error?│────▶│Create Secret API│─────┐
```

Comment:
Same formatting, just a couple more spaces or dashes? :D It's REALLY awesome

Reply:
Please take a screenshot of what you especially want. I only partially understand hahaha